nvflare/app_opt/lightning/api.py

```diff
@@ -22,10 +22,12 @@
 from nvflare.client.api import _get_model_registry, clear, get_config, init, receive, send
 from nvflare.client.config import ConfigKey
 
+from .callbacks import RestoreState
+
 FL_META_KEY = "__fl_meta__"
 
 
-def patch(trainer: pl.Trainer):
+def patch(trainer: pl.Trainer, restore_optimizers: bool = True):
```

Comment:
Should this default be True or False?

What does 80% cases need?

Do we usually want to cache these states or not?

@holgerroth  @ZiyueXu77

Reply:
I’m proposing `True` by default, as you need it for lr scheduling. Also, all our existing Learner-based implementations are keeping the state of the local optimizer. So, with `True`, you should be able to get the same behavior as with Learner-based PyTorch implementations.